### ISSUE TITLE
AuthV2: HTTP-500 for staff

### ISSUE THREAD

**Issue by kiblik** [DefectDojo/django-DefectDojo#4677]
**Be informative**
I tried to access product as staff but I received `HTTP-500`

**Bug description**
- Browser: `HTTP-500`
- Console: `AttributeError: 'NoneType' object has no attribute 'id'`

**Steps to reproduce**
1. Use clean DD
2. Create product
3. Create user - active, staff, (no superuser, no additional permissions)
4. Log in as a new user
5. View product
6. See log

**Expected behavior**
Show product

**Deployment method**
- Docker

**Environment information**
 - DefectDojo Commit Message: `[2021-06-17 18:22:24 +0200] 08b3335ba: fix: missing Yarn Audit support in documentation (#4669) [ (HEAD, upstream/dev)]`

**Console logs**
```python
uwsgi_1         | [17/Jun/2021 19:31:15] ERROR [django.request:224] Internal Server Error: /product/1
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
uwsgi_1         |     response = get_response(request)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
uwsgi_1         |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
uwsgi_1         |   File "/app/./dojo/authorization/authorization_decorators.py", line 39, in _wrapped
uwsgi_1         |     return func(request, *args, **kwargs)
uwsgi_1         |   File "/app/./dojo/product/views.py", line 139, in view_product
uwsgi_1         |     product_members = get_authorized_members_for_product(prod, Permissions.Product_View)
uwsgi_1         |   File "/app/./dojo/product/queries.py", line 70, in get_authorized_members_for_product
uwsgi_1         |     if user.is_superuser or user_has_permission(user, product, permission):
uwsgi_1         |   File "/app/./dojo/authorization/authorization.py", line 17, in user_has_permission
uwsgi_1         |     if hasattr(user, 'global_role') and role_has_permission(user.global_role.role.id, permission):
uwsgi_1         | AttributeError: 'NoneType' object has no attribute 'id'
uwsgi_1         | ERROR:django.request:Internal Server Error: /product/1
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
uwsgi_1         |     response = get_response(request)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
uwsgi_1         |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
uwsgi_1         |   File "/app/./dojo/authorization/authorization_decorators.py", line 39, in _wrapped
uwsgi_1         |     return func(request, *args, **kwargs)
uwsgi_1         |   File "/app/./dojo/product/views.py", line 139, in view_product
uwsgi_1         |     product_members = get_authorized_members_for_product(prod, Permissions.Product_View)
uwsgi_1         |   File "/app/./dojo/product/queries.py", line 70, in get_authorized_members_for_product
uwsgi_1         |     if user.is_superuser or user_has_permission(user, product, permission):
uwsgi_1         |   File "/app/./dojo/authorization/authorization.py", line 17, in user_has_permission
uwsgi_1         |     if hasattr(user, 'global_role') and role_has_permission(user.global_role.role.id, permission):
uwsgi_1         | AttributeError: 'NoneType' object has no attribute 'id'
```

**Comment by damiencarol** [comment]
@StefanFl

**Comment by kiblik** [comment]
Fixed in #4650
